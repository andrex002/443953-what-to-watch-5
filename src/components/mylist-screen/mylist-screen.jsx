import React, {useEffect} from "react";
import PropTypes from "prop-types";
import FilmCardsList from "../film-cards-list/film-cards-list";
import LogoHeader from "../logo-header/logo-header";
import UserBlock from "../user-block/user-block";
import PageFooter from "../page-footer/page-footer";
import withActiveCard from "../../hocs/with-active-card/with-active-card";
import {connect} from "react-redux";
import {fetchFavoriteFilms} from "../../store/api-action";

const FilmCardsListWrapped = withActiveCard(FilmCardsList);

const MyListScreen = (props) => {
  const {favoriteFilms, onFilmCardClick, getFavoriteFilms} = props;

  useEffect(() => {
    getFavoriteFilms();
  }, []);

  return (
    <div className="user-page">
      <header className="page-header user-page__head">

        <LogoHeader />

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock />

      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        <FilmCardsListWrapped films={favoriteFilms} onFilmCardClick={onFilmCardClick} />

      </section>

      <PageFooter />

    </div>
  );
};

MyListScreen.propTypes = {
  favoriteFilms: PropTypes.array.isRequired,
  onFilmCardClick: PropTypes.func.isRequired,
  getFavoriteFilms: PropTypes.func.isRequired
};

const mapStateToProps = ({DATA}) => ({
  favoriteFilms: DATA.favoriteFilms
});

const mapDispatchToProps = (dispatch) => ({
  getFavoriteFilms() {
    dispatch(fetchFavoriteFilms());
  }
});

export {MyListScreen};
export default connect(mapStateToProps, mapDispatchToProps)(MyListScreen);
