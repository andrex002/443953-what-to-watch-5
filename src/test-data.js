export const film = {
  "id": 1,
  "title": "The Grand Budapest Hotel",
  "image": "img/the-grand-budapest-hotel-poster.jpg",
  "previewImage": "img/the-grand-budapest-hotel.jpg",
  "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
  "bgColor": "#ffffff",
  "srcVideo": "https://some-link",
  "video": "https://some-link",
  "previewVideo": "https://some-link",
  "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
  "rating": {
    "score": 8.9,
    "count": 240
  },
  "director": "Wes Andreson",
  "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
  "duration": 99,
  "genre": "Comedy",
  "year": 2014,
  "isFavorite": false
};

export const films = [
  {
    "id": 1,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  },
  {
    "id": 2,
    "title": "The Grand Budapest Hotel",
    "image": "img/the-grand-budapest-hotel-poster.jpg",
    "previewImage": "img/the-grand-budapest-hotel.jpg",
    "bgImage": "img/the-grand-budapest-hotel-bg.jpg",
    "bgColor": "#ffffff",
    "video": "https://some-link",
    "previewVideo": "https://some-link",
    "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
    "rating": {
      "score": 8.9,
      "count": 240
    },
    "director": "Wes Andreson",
    "actors": ["Bill Murray", "Edward Norton", "Jude Law", "Willem Dafoe", "Saoirse Ronan"],
    "duration": 99,
    "genre": "Comedy",
    "year": 2014,
    "isFavorite": false
  }
];

export const comment = {
  "id": 1,
  "userId": 4,
  "userName": "Kate Muir",
  "rating": 8.9,
  "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
  "date": "2019-05-08T14:13:56.569Z"
};

export const comments = [
  {
    "id": 1,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
  {
    "id": 2,
    "userId": 4,
    "userName": "Kate Muir",
    "rating": 8.9,
    "textComment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
    "date": "2019-05-08T14:13:56.569Z"
  },
];

export const defaultState = {
  DATA: {
    allFilms: films,
    filteredFilms: films,
    promoFilm: film,
    currentFilm: film,
    favoriteFilms: films,
    comments: comments,
    numberFilmsShown: 8,
    isCommentSending: false,
    isCommentSendError: false
  },
  FILTER: {
    activeGenre: `All genres`,
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    name: `userName`,
    avatar: `avatar.jpg`
  }
};

export const noop = () => {};