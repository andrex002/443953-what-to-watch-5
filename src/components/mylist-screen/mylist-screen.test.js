import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {MyListScreen} from "./mylist-screen";
import {films, noop} from  "../../test-data";

jest.mock(`../user-block/user-block`, () => `UserBlock`);

it(`Should MyListScreen render correctly`, () => {
  const tree = renderer.create(
    <BrowserRouter>
      <MyListScreen
        favoriteFilms={films}
        onFilmCardClick={noop}
        getFavoriteFilms={noop}
      />
    </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

