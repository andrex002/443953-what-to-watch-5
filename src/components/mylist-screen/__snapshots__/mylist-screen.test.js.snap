// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyListScreen render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <UserBlock />
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        data-id={1}
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="175"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            The Grand Budapest Hotel
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        data-id={2}
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="175"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            The Grand Budapest Hotel
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
